src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs

```diff
@@ -423,5 +385,44 @@ protected override void OnElementChanged(ElementChangedEventArgs<MediaElement> e
 		}
 
 		protected virtual void UpdateBackgroundColor() => BackgroundColor = Element.BackgroundColor.ToUIColor();
+
+		protected override void Dispose(bool disposing)
```

Comment:
Dispose is called once (or twice)
Can't we get a scenario where OnElementChanged is called multiple times from the view (Sharing Renderer among different Elements). Probably no, at least I've never seen it :)

But in such scenario, we will get memory leak, because we subscribe on events and create IDisposable observer there

 ```csharp
				Element.PropertyChanged += OnElementPropertyChanged;
				Element.SeekRequested += MediaElementSeekRequested;
				Element.StateRequested += MediaElementStateRequested;
				Element.PositionRequested += MediaElementPositionRequested;
```

Reply:
@AndreiMisiukevich. So we should keep the unsubscribe code inside the `e.OldElement != null` or on both? I don't if that throws an exception if we don't have more events to unsubscribe.

Reply:
Don't you think it makes sense to preserve the logic where it exists now?

Reply:
To me make more sense to clean up resources on the Dispose method, because we can prevent the `System.ObjectDisposedException` more easily. Anyway, I'm ok to move the code to the current location

Reply:
Renderers are designed to be possible to share a single renderer among several Elements.
But I am not sure that this feature is used. (One element - one renderer.)

@jsuarezruiz @sthewissen ?